mrp_multi_level/wizards/mrp_multi_level.py

```diff
@@ -200,45 +211,39 @@ def explode_action(
         mrp_date_demand = mrp_action_date
         if mrp_date_demand < date.today():
             mrp_date_demand = date.today()
-        if not product_mrp_area_id.product_id.bom_ids:
+        bom = self._get_bom_to_explode(product_mrp_area_id)
+        if not bom:
             return False
-        bomcount = 0
-        for bom in product_mrp_area_id.product_id.bom_ids:
-            if not bom.active or not bom.bom_line_ids:
+        for bomline in bom.bom_line_ids:
```

Comment:
we can use `bom.bom_line_ids.filtered(lambda r: r.product_qty < 0 or r.product_id.type != "product")`

Reply:
Yes, that's true! 

Well, it really is `bom.bom_line_ids.filtered(lambda r: r.product_qty >= 0 or r.product_id.type != "product")`, but thanks!